pwd_guess.py

```diff
@@ -676,8 +696,10 @@ def prepare_x_data(self, x_strs):
     def prepare_y_data(self, y_str_list):
         y_vec = np.zeros((len(y_str_list), self.ctable.vocab_size),
                          dtype=np.bool)
-        self.ctable.y_encode_into(y_vec, y_str_list)
-        #todo: check y_encode_into or encode_into
+        if self.config.sequence_model == Sequence.MANY_TO_ONE:
```

Comment:
Instead of adding a conditional check, could you move the path for the many to many model into the subclass?